### PR TITLE
OEL 664: Banner paragraph

### PR DESCRIPTION
**Pull request by escuriola**
Jira issue(s):
- https://citnet.tech.ec.europa.eu/CITnet/jira/browse/OEL-664

### CODE REVIEW

File: bcl-builder.config.js [hunk 1]
```diff
@@ -44,6 +44,11 @@ module.exports = {
       to: path.resolve(outputFolder, "assets/js"),
       options: { up: true },
     },
+    {
```
Comment (escuriola) [inline bcl-builder.config.js]: Temporal until upgrade to BCL 0.12.0

File: templates/patterns/banner/banner.html.twig [hunk 2]
```diff
@@ -0,0 +1,46 @@
+{% spaceless %}
```
Comment (escuriola) [inline templates/patterns/banner/banner.html.twig]: Temporal until BCL 0.12.0

File: src/ValueObject/DateValueObject.php [hunk 3]
```diff
@@ -0,0 +1,167 @@
+<?php
```
Comment (drishu) [inline src/ValueObject/DateValueObject.php]: remove this

File: src/ValueObject/DateValueObjectInterface.php [hunk 4]
```diff
@@ -0,0 +1,85 @@
+<?php
```
Comment (drishu) [inline src/ValueObject/DateValueObjectInterface.php]: remove this

File: src/ValueObject/FileValueObject.php [hunk 5]
```diff
@@ -0,0 +1,267 @@
+<?php
```
Comment (drishu) [inline src/ValueObject/FileValueObject.php]: remove this

File: src/ValueObject/GalleryItemValueObject.php [hunk 6]
```diff
@@ -0,0 +1,187 @@
+<?php
```
Comment (drishu) [inline src/ValueObject/GalleryItemValueObject.php]: remove this

File: templates/paragraphs/paragraph--oe-banner.html.twig [hunk 7]
```diff
@@ -0,0 +1,42 @@
+{#
+/**
+ * @file
+ * Theme override to display the 'quote' type paragraph.
```
Comment (drishu) [inline templates/paragraphs/paragraph--oe-banner.html.twig]:
```suggestion
 * Theme override to display the 'banner' type paragraph.
```

File: bcl-builder.config.js (hunk 1 above)
Comment (drishu) [reply bcl-builder.config.js]: remove this

File: templates/patterns/banner/banner.html.twig (hunk 2 above)
Comment (drishu) [reply templates/patterns/banner/banner.html.twig]: move the contents to pattern-banner.html.twig and remove this file

File: templates/patterns/banner/pattern-banner.html.twig [hunk 8]
```diff
@@ -0,0 +1,25 @@
+{#
```
Comment (drishu) [inline templates/patterns/banner/pattern-banner.html.twig:1]: here add the contents of banner.html.twig (you can keep lines 7 to 14

File: templates/patterns/banner/pattern-banner--variant-primary.html.twig [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+#}
+{% set attributes = attributes.addClass([
+    'bg-primary',
+    'text-white',
+  ])
+%}
+{% if centered  %}
+  {% set attributes = attributes.addClass(['text-center'])%}
+{% endif %}
+{% include '@oe-bcl/bcl-banner/banner.html.twig' with {
```
Comment (drishu) [inline templates/patterns/banner/pattern-banner--variant-primary.html.twig]: extend pattern-banner.html.twig use block content (or as many blocks as you need) to do your alterations

File: templates/patterns/banner/pattern-banner--variant-image_shade.html.twig [hunk 10]
```diff
@@ -0,0 +1,20 @@
+{#
+/**
+ * @file
+ * Banner component.
+ */
+#}
+{% if centered  %}
+  {% set attributes = attributes.addClass(['text-center'])%}
+{% endif %}
+{% include '@oe-bcl/bcl-banner/banner.html.twig' with {
```
Comment (drishu) [inline templates/patterns/banner/pattern-banner--variant-image_shade.html.twig]: extend pattern-banner.html.twig use block content (or as many blocks as you need) to do your alterations

File: templates/patterns/banner/pattern-banner.html.twig [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+/**
+ * @file
+ * Banner component.
```
Comment (drishu) [inline templates/patterns/banner/pattern-banner.html.twig]:
```suggestion
 * Banner pattern.
```

**Review by drishu — CHANGES_REQUESTED** [review]

**Review by drishu — APPROVED** [review]

**Review by GilNovacomm — APPROVED** [review]
Internal URL works as expected.

**Pull request merged**